Anvil/Tools/Striker.pm

```diff
@@ -97,75 +97,178 @@ sub check_httpd_conf
+		$write_shell_call .= "set ".$last_rewritecond_arg."[2] -f\n";
+		$write_shell_call .= "set ".$new_directive." ".$rewriterule_directive."\n";
+		$write_shell_call .= "set ".$last_rewriterule_arg."[1] ![^/]+[.]html\$\n";
+		$write_shell_call .= "set ".$last_rewriterule_arg."[2] %{REQUEST_FILENAME}.html\n";
+		$write_shell_call .= "set ".$last_rewriterule_arg."[3] [L]\n";
+	}
+
+	if ($is_write)
+	{
+		$write_shell_call .= "save\nquit\nEOF\n";
+
+		($shell_output, $shell_return_code) = $anvil->System->call({ shell_call => $write_shell_call });
+		$anvil->Log->variables({ source => $THIS_FILE, line => __LINE__, level => $debug, list => {
+			shell_call        => $write_shell_call,
+			shell_output      => $shell_output,
+			shell_return_code => $shell_return_code
+		} });
+
+		my $new_httpd_conf = $anvil->data->{path}{data}{httpd_conf}.".augnew";
+
+		if (not -s $new_httpd_conf)
 		{
-			$in_dir_module = 1;
-			$anvil->Log->variables({source => $THIS_FILE, line => __LINE__, level => $debug, list => { in_dir_module => $in_dir_module }});
+			# No new conf file created; abort.
+			return(0);
 		}
-		if ($in_dir_module)
+
+		my $diff_output = diff $anvil->data->{path}{data}{httpd_conf}, $new_httpd_conf, { STYLE => 'Unified' };
+		$anvil->Log->variables({ source => $THIS_FILE, line => __LINE__, level => $debug, list => { diff_output => $diff_output } });
+
+		# Test the new config file before overwriting.
+		$read_shell_call = $anvil->data->{path}{exe}{httpd}." -t -f ".$new_httpd_conf;
+
+		($shell_output, $shell_return_code) = $anvil->System->call({ shell_call => $read_shell_call });
+		$anvil->Log->variables({ source => $THIS_FILE, line => __LINE__, level => $debug, list => {
+			shell_call        => $read_shell_call,
+			shell_output      => $shell_output,
+			shell_return_code => $shell_return_code
+		} });
+
+		if ($shell_return_code == 0)
 		{
-			if ($line =~ /^<\/IfModule>/)
-			{
-				$in_dir_module = 0;
-				$anvil->Log->variables({source => $THIS_FILE, line => __LINE__, level => $debug, list => { in_dir_module => $in_dir_module }});
-			}
-			elsif ($line =~ /^\s+DirectoryIndex (.*)/)
-			{
-				my $directory_index = $1;
-				$anvil->Log->variables({source => $THIS_FILE, line => __LINE__, level => $debug, list => { directory_index => $directory_index }});
-				if ($directory_index ne "cgi-bin/striker")
-				{
-					$line        =~ s/$directory_index/cgi-bin\/striker/;
-					$update_file =  1;
-					$anvil->Log->variables({source => $THIS_FILE, line => __LINE__, level => $debug, list => { 
-						line        => $line,
-						update_file => $update_file, 
-					}});
-				}
-			}
+			# No issues found during the test; proceed to overwriting.
+
+			$anvil->Storage->backup({ debug => $debug, file => $anvil->data->{path}{data}{httpd_conf} });
+
+			$write_shell_call = $anvil->data->{path}{exe}{mv}." -f ".$new_httpd_conf." ".$anvil->data->{path}{data}{httpd_conf};
```

Comment:
There's a Storage->backup() method available (not sure if it would have been useful here)

Reply:
Also, you can specify to automatically create a backup when writing a file that already exists (overwrite => 1, backup => 1, iirc).